docker/docker-compose.yml

```diff
@@ -0,0 +1,248 @@
+version: '3'
+services:
+  gobblin:
+    container_name: gobblin
+    hostname: gobblin
+    image: pnda/gobblin:0.11.0-0.1.0
+    environment:
+    - HDFS_URL=hdfs://hdfs-namenode:8020
+    - MASTER_DATASET_DIRECTORY=/user/pnda/PNDA_datasets/datasets
+    - MASTER_DATASET_QUARANTINE_DIRECTORY=/user/pnda/PNDA_datasets/quarantine
+    - MAX_MAPPERS=4
+    - KAFKA_BROKERS=kafka:9092
+  spark-master:
+    container_name: spark-master
+    hostname: spark-master
+    image: gradiant/spark:2.3.0
+    entrypoint:
+    - /opt/spark/sbin/start-master.sh
+    environment:
+    - SPARK_MASTER_HOST=spark-master
+    - SPARK_MASTER_PORT=7077
+  spark-slave:
+    container_name: spark-slave
+    hostname: spark-slave
+    image: repo.gradiant.org:9011/spark:2.3.0
+    entrypoint:
+    - /opt/spark/sbin/start-slave.sh
+    - spark://spark-master:7077
+  jupyter:
+    container_name: jupyter
+    hostname: jupyter
+    image: pnda/jupyter:4.4.0
+    volumes:
+    - jupyter-home:/home
+    environment:
+    - SPARK_MASTER_URL=spark://spark-master:7077
+    - HADOOP_DISTRO=env
+    - HDFS_ROOT_URI=hdfs://hdfs-namenode:8020
+  jupyter-ssh:
+    container_name: jupyter-ssh
+    image: gradiant/openssh-server
+    network_mode: service:jupyter
+    volumes:
+    - jupyter-home:/home
+    depends_on:
+    - jupyter
+  grafana:
+    container_name: grafana
+    hostname: grafana
+    image: grafana/grafana:5.0.3
+    environment:
+    - GF_SECURITY_ADMIN_USER=pnda
+    - GF_SECURITY_ADMIN_PASSWORD=pnda
+    ports:
+    - 3000:3000
+    volumes:
+    - grafana:/var/lib/grafana
+  deployment-manager:
+    container_name: deployment-manager
+    hostname: deployment-manager
+    image: pnda/deployment-manager:1.0.0
+    environment:
+    - JUPYTER_HOST=jupyter
+    - DATA_LOGGER_URL=http://console-backend:3001 #data-logger uses the data-manager network stack
+    - PACKAGE_REPOSITORY_URL=http://package-repository:8888
+    - HADOOP_DISTRO=env
+    - HDFS_ROOT_URI=hdfs://hdfs-namenode
+    - SPARK_MASTER=spark://spark-master:7077
+    - YARN_NODE_MANAGERS=localhost
+    - ZOOKEEPER_QUORUM=zookeeper
+    - LOG_LEVEL=DEBUG
+    volumes:
+    - dm_keys:/opt/pnda/dm_keys
+  kafka-manager:
+    container_name: kafka-manager
+    hostname: kafka-manager
+    image: gradiant/kafka-manager:1.3.3.17
```

Comment:
Would it be possible to either include the DockerFile in the pndaproject organisation and/or put more details about them in the README so that user can easily access to them. Just in case they want to review/build them.
Or you can include a build in the service but not sure this is the best option.
Maybe better to use variable substitution for all the images so once you start the script, you load a set of image name defined in a configuration file, so that people could easily change them if they are currently working on Red PNDA no?

Reply:
Regarding "variable substitution" option for the images, I think the best place to change them if you are working on Red PNDA is to directly edit the docker-compose.yml and customize it for you.
Or if you want to explicitly have a place to put only image versions, I can create a docker-compose [.env file](https://docs.docker.com/compose/environment-variables/#the-env-file).